src/demo/DemoTestRepaymentService.java

```diff
@@ -215,4 +212,126 @@ public static void testGetInvestorAccountRepayments() throws MambuApiException {
 		}
 
 	}
+
+	// Tests deleting a repayment for a loan account
+	public static void testDeleteRepayment() throws MambuApiException {
+
+		String methodName = new Object() {}.getClass().getEnclosingMethod().getName();
+		System.out.println(methodName = "\nIn " + methodName);
+
+		if (!demoLoanAccount.getState().equals(AccountState.PENDING_APPROVAL)) {
+			System.out.println("Invalid account state for " + methodName);
+			return;
+		}
+
+		// get repayments for the loan account
+		List<Repayment> repayments = testGetLoanAccountRepayments();
+		if (repayments.size() >= 2) {
+			List<Repayment> notPaidRepayments = filterNotPaidRepayments(repayments);
+
+			// edits the schedule
+			reduceScheduleByFirstRepayment(notPaidRepayments);
+
+			List<Repayment> returnedRepayments = submitEditedRepaymentsToMambu(notPaidRepayments);
+
+			// delete first of the repayments
+			RepaymentsService repaymentService = MambuAPIFactory.getRepaymentsService();
+			Boolean deleted = repaymentService.deleteLoanRepayment(LOAN_ACCOUNT_ID,
+					returnedRepayments.get(0).getEncodedKey());
+
+			System.out.println("Repayment was successfully deleted = " + deleted);
+		} else {
+			System.out.println(methodName + "can`t be performed due e to repayments no.");
+			System.out.println("There should be at least 2 repayments");
+		}
+	}
+
+	/**
+	 * Submits the list of updated repayments received as parameter to this method in order to update the schedule of a
+	 * loan.
+	 * 
+	 * @param editedRepayments
+	 *            a list of updated repayments to be sent to Mambu.
+	 * 
+	 * @return a list of updated repayments from Mambu
+	 * 
+	 * @throws MambuApiException
+	 */
+	private static List<Repayment> submitEditedRepaymentsToMambu(List<Repayment> editedRepayments)
+			throws MambuApiException {
+
+		// Submit Update schedule API request with these updated entries
+		JSONLoanRepayments loanRepayments = new JSONLoanRepayments(editedRepayments);
+		RepaymentsService repaymentService = MambuAPIFactory.getRepaymentsService();
+		System.out.println("Updating Repayments schedule for Loan Account ID=" + LOAN_ACCOUNT_ID + "\tRepayments ="
+				+ editedRepayments.size());
+
+		List<Repayment> updatedRepayments = repaymentService.updateLoanRepaymentsSchedule(LOAN_ACCOUNT_ID,
+				loanRepayments);
+
+		int totalReturned = (updatedRepayments == null) ? 0 : updatedRepayments.size();
+		System.out.println("Total Repayments returned after update=" + totalReturned);
+		// Can also see detailed update log on a Dashboard in Mambu
+		return updatedRepayments;
+	}
+
+	/**
+	 * Edits the list of not paid yet repayments by setting the first one`s principal to be "0"
+	 * 
+	 * @param notPaidRepayments
+	 *            the list of not paid repayments to be adjusted.
+	 */
+	private static void reduceScheduleByFirstRepayment(List<Repayment> notPaidRepayments) {
```

Comment:
In order to avoid the potential confusion that some users of this method might encounter, maybe it would be better to have this method returning a new list of repayments instead of changing the list that is given as an argument.